Close coverage gaps identified in review of PR #823

wolenetz opened the pull request:
1. addSourceBuffer(null) should result in TypeError (not
   NotSupportedError as currently expected by
   mediasource-addsourcebuffer.html)
   Note, https://crbug.com/623781 tracks fixing the Chrome M54
   compliance issue behind at least the failure of 2 TypeError subtests
   in mediasource-addsourcebuffer.html.
2. isTypeSupported(mime) returning true should mean that
   canPlayType(mime) is either "maybe" or "probably".
3. Explicitly test isTypeSupported for each of: "xxx", "text/html",
   "image/jpeg" result in false.
4. Explicitly test sourceBuffers added to distinct MediaSource objects
   give error when they are used in removeSourceBuffer to the wrong
   MediaSource. This can be added to mediasource-removesourcebuffer.html

This commit also removes a duplicated invalid type test from
mediasource-is-type-supported.html.

wolenetz committed:
Close coverage gaps identified in review of PR #823

1. addSourceBuffer(null) should result in TypeError (not
   NotSupportedError as currently expected by
   mediasource-addsourcebuffer.html)
   Note, https://crbug.com/623781 tracks fixing the Chrome M54
   compliance issue behind at least the failure of 2 TypeError subtests
   in mediasource-addsourcebuffer.html.
2. isTypeSupported(mime) returning true should mean that
   canPlayType(mime) is either "maybe" or "probably".
3. Explicitly test isTypeSupported for each of: "xxx", "text/html",
   "image/jpeg" result in false.
4. Explicitly test sourceBuffers added to distinct MediaSource objects
   give error when they are used in removeSourceBuffer to the wrong
   MediaSource. This can be added to mediasource-removesourcebuffer.html

This commit also removes a duplicated invalid type test from
mediasource-is-type-supported.html.

wolenetz commented:
@jdsmith3000 Please take a look. Thanks!

jdsmith3000 commented on media-source/mediasource-is-type-supported.html:
```diff
@@ -76,6 +82,17 @@
               'video/mp4;codecs="avc1.4d001e,mp4a.40.5"',
           ], true, 'Test valid MP4 type');
+
+          mediasource_test(function(test, mediaElement, mediaSource)
```
This test enforces a non-normative note and doesn't seem appropriate.

I agree with the note in principle, but know of at least one instance in Edge (VP9) where there is only MSE support.  That was done intentionally in part because Edge doesn't currently support Vorbis, and many progressive download uses of VP9 are paired with Vorbis for audio.

wolenetz replied:
I understand, and that is an interesting datapoint. I assume that making Edge return "maybe" for canPlayType(... vorbis) might also not be a good idea, since apps may take that as a signal to attempt progressive playback of media including vorbis. I'll remove this MSE suite test, regardless, though in general, that non-normative note should be adhered to.

jdsmith3000 commented:
I added one comment about isTypeSupported checking against canPlayType.

Otherwise, this LGTM.  The multiple SourceBuffer test looks clean and useful.

wolenetz committed:
Remove a test of non-normative note contents

Removes the test added in 21b3020d where isTypeSupported(mime) returning
true should mean that canPlayType(mime) is either "maybe" or "probably".
Such a test is invalid on some platforms where MSE may support more
codecs than regular src= playback.
See https://github.com/w3c/web-platform-tests/pull/3635#discussion_r77712770

wolenetz commented:
Thanks for review. I removed the offending test and will merge this shortly.
